src/cli/cli.js

```diff
@@ -124,24 +125,29 @@ async function runTests (argParser) {
 }
 
 async function listBrowsers (providerName) {
-    const provider = await browserProviderPool.getProvider(providerName);
 
-    if (!provider)
-        throw new GeneralError(RUNTIME_ERRORS.browserProviderNotFound, providerName);
-
-    if (provider.isMultiBrowser) {
-        const browserNames = await provider.getBrowserList();
+    let browserNames;
 
+    try {
+        browserNames = await listBrowsersApi(providerName);
+    }
+    catch (e) {
         await browserProviderPool.dispose();
-
-        if (providerName === 'locally-installed')
-            console.log(browserNames.join('\n'));
-        else
-            console.log(browserNames.map(browserName => `"${providerName}:${browserName}"`).join('\n'));
+        throw e.code === RUNTIME_ERRORS.browserProviderNotFound
```

Comment:
I think we can just throw a `GeneralError` in `listBrowsers` and remove this `catch` clause completely.